bindings/python/pymongocrypt/mongocrypt.py

```diff
@@ -366,7 +388,130 @@ def __init__(self, ctx, command):
+            form { "v" : BSON value to encrypt }}.
+          - `opts`: A :class:`ExplicitEncryptOpts`.
+        """
+        super(ExplicitEncryptionContext, self).__init__(ctx)
+        try:
+            if not lib.mongocrypt_ctx_setopt_algorithm(
+                    ctx, opts.algorithm, -1):
+                self._raise_from_status()
+
+            if opts.key_id is not None:
+                with MongoCryptBinaryIn(opts.key_id) as binary:
+                    if not lib.mongocrypt_ctx_setopt_key_id(ctx, binary.bin):
+                        self._raise_from_status()
+
+            if opts.key_alt_name is not None:
+                with MongoCryptBinaryIn(opts.key_alt_name) as binary:
+                    if not lib.mongocrypt_ctx_setopt_key_alt_name(ctx,
+                                                                  binary.bin):
+                        self._raise_from_status()
+
+            with MongoCryptBinaryIn(value) as binary:
+                if not lib.mongocrypt_ctx_explicit_encrypt_init(ctx,
+                                                                binary.bin):
+                    self._raise_from_status()
+        except Exception:
+            # Destroy the context on error.
+            self._close()
+            raise
+
+
+class ExplicitDecryptionContext(MongoCryptContext):
+    __slots__ = ()
+
+    def __init__(self, ctx, value):
+        """Abstracts libmongocrypt's mongocrypt_ctx_t type.
+
+        :Parameters:
+          - `ctx`: A mongocrypt_ctx_t. This MongoCryptContext takes ownership
+            of the underlying mongocrypt_ctx_t.
+          - `value`: The encoded BSON value to decrypt.
+        """
+        super(ExplicitDecryptionContext, self).__init__(ctx)
+
+        try:
+            with MongoCryptBinaryIn(value) as binary:
+                if not lib.mongocrypt_ctx_explicit_decrypt_init(ctx,
+                                                                binary.bin):
+                    self._raise_from_status()
+        except Exception:
+            # Destroy the context on error.
+            self._close()
+            raise
+
+
+class DataKeyContext(MongoCryptContext):
+    __slots__ = ()
+
+    def __init__(self, ctx, kms_provider, opts):
+        """Abstracts libmongocrypt's mongocrypt_ctx_t type.
+
+        :Parameters:
+          - `ctx`: A mongocrypt_ctx_t. This MongoCryptContext takes ownership
+            of the underlying mongocrypt_ctx_t.
+          - `kms_provider`: The KMS provider.
+          - `opts`: An optional class:`DataKeyOpts`.
+        """
+        super(DataKeyContext, self).__init__(ctx)
+        try:
+            if kms_provider == 'aws':
+                if opts is None or opts.master_key is None:
+                    raise ValueError('opts.master_key is required for '
+                                     'kms_provider: "aws"')
+                if 'region' not in opts.master_key or 'key' not in \
```

Comment:
Use parens, not \ for line continuation.

Reply:
Fixed.